### ISSUE TITLE
Unable to Add header with IClient

### ISSUE THREAD

**Issue by kikshore** [commercetools/commercetools-dotnet-core-sdk-v2#262]
I am using IClient to make a graphQL request like this and trying to set the x-correlation-id header  that is passed from the front end with the graphql request. This code is in my web service.

//I hand typed this code so please ignore typos
```
var correlationId = req.Headers.GetValues("x-correlation-id").FirstOrDefault();

ApiHttpHeaders headers = new ApiHttpHeaders();
headers.AddHeader("X-Correlation-ID" , correlationId);

var myObject = await _client.
                                   .WithApi()
                                   .WithProjectKey(projectKey)
                                   .Graphql()
                                   .Post(graphqlRequest)
                                   .WithHeaders(headers)   // THIS THROWS AN ERROR
                                   .ExecuteAsync()
                                   .ConfigureAwait(false);
```
                                   
                                   
                                   
ERROR : {"statusCode":400 , "message" : "Invalid X-Correlation-ID header . It may only contain alphanumeric characters , underscores and hyphens and have a length of 8 to 256 characters.: , "errors" :[{"code":"InvalidInput" , "message":"Invalid X-Correlation-ID header . It may only contain alphanumeric characters , underscores and hyphens and have a length of 8 to 256 characters."}]}

So how do we send the x-correlation-id header with IClient.

KIKSH

**Comment by jenschude** [comment]
Sorry for the late answer.

It's exactly how you can set the header. The issue is that the API has constraints for the correlation ID header

Please see https://docs.commercetools.com/api/general-concepts#correlation-id

**Comment by kikshore** [comment]
No problem , Thanks @jenschude  . I  tried using different values for the correlation Id starting from using GUID to a basic string with just alphabets . 

Can you share some more info on the constraints .  Or any example that you may have.

**Comment by vndteksysrxa** [comment]
Did anyone find the solution to this ? Though we are using proper correlation id that satisfy the condition I am also getting the same error.

**Comment by jenschude** [comment]
I did a test and realized that by adding a correlationId header to the request the client would send two correlation ids and it seems that the API doesn't accept two correlation id headers.

I opened a PR which checks if a request already has this header set before adding a custom ID and also allows to replace the provider. See #305

**Comment by jenschude** [comment]
I just released 11.6.0 with the fix and support for custom correlation id providers
